presto-native-execution/presto_cpp/main/types/tests/PrestoToVeloxQueryPlanTest.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/types/PrestoToVeloxQueryPlan.h"
+#include <gtest/gtest.h>
+#include "velox/common/base/tests/GTestUtils.h"
+
+using namespace facebook::presto;
+using namespace facebook::velox;
+
+class PrestoToVeloxQueryPlanTest : public ::testing::Test {};
+
+TEST_F(PrestoToVeloxQueryPlanTest, parseSqlFunctionHandleWithZeroParam) {
+  std::string str = R"(
+          {
+            "@type": "call",
+            "arguments": [
+              {
+                "@type": "variable",
+                "name": "event_based_revenue",
+                "type": "real"
+              }
+            ],
+            "displayName": "sum",
+            "functionHandle": {
+              "@type": "json_file",
+              "functionId": "json.x4.sum;",
+              "version": "1"
+            },
+            "returnType": "double"
+          }
+    )";
+
+  json j = json::parse(str);
+  facebook::presto::protocol::CallExpression p = j;
+  ASSERT_NE(p.functionHandle, nullptr);
+
+  std::vector<TypePtr> actualRawInputTypes;
+  std::vector<TypePtr> expectedRawInputTypes;
+  TypeParser typeParser;
+  auto sqlFunctionHandle =
```

Comment:
I blindly copied the test cases from https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/presto_protocol/tests/CallExpressionTest.cpp. 
I just needed to have my hands on an instance of a `SqlFunctionHandle` for the unit test, because the `parseSqlFunctionHandle` function had a bug which failed in the parsing of the `functionId` string.
Do you think this is a good way to test it? Please let me know if you have any better ideas.

Reply:
@pdabre12 : Where do you anticipate this json to be generated ? You can run a query with an aggregate function and pick up the json from the web-console.

Reply:
It will only be generated for functions in the new function namespace manager : https://github.com/prestodb/presto/pull/23358. 
Below is the json for the `array_constructor` function :  

```
                "@type" : "call",
                  "sourceLocation" : {
                    "line" : 1,
                    "column" : 74
                  },
                  "displayName" : "array_constructor",
                  "functionHandle" : {
                    "@type" : "native",
                    "signature" : {
                      "name" : "native.default.array_constructor",
                      "kind" : "SCALAR",
                      "typeVariableConstraints" : [ ],
                      "longVariableConstraints" : [ ],
                      "returnType" : "array(double)",
                      "argumentTypes" : [ "double" ],
                      "variableArity" : false
                    },
                    "functionId" : "native.default.array_constructor;double",
                    "version" : "1"
                  },
```
The problem is without https://github.com/prestodb/presto/pull/23358 landing, native execution does not understand the `native` functionHandle type, I suppose I could pick the json but keep the functionHandle type as `json_file` as we really only are testing the  parsing of the functionId string because that's where the bug manifests, let me know if that sounds okay.

Reply:
@pdabre12 : Yes, that is fine. Thanks.